alluxio-ui/common/src/components/FileView/FileView.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {faCheckSquare, faSquare} from '@fortawesome/free-regular-svg-icons'
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
+import React from 'react';
+import {Link} from 'react-router-dom';
+import {Button, ButtonGroup, Form, FormGroup, Input, Label} from 'reactstrap';
+
+import {IFileInfo} from '../../constants';
+
+interface IFileViewProps {
+  allowDownload?: boolean;
+  beginInputHandler: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  end?: string
+  endInputHandler: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  lastFetched: {
+    end?: string;
+    limit?: string;
+    offset?: string;
+    path?: string;
+  };
+  limit?: string;
+  offset?: string;
+  offsetInputHandler: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  path?: string;
+  queryStringPrefix: string;
+  queryStringSuffix: string;
+  textAreaHeight?: number;
+  viewData: {
+    currentDirectory?: IFileInfo;
+    'currentPath': string;
+    'debug': boolean;
+    'fatalError': string;
+    'fileData': string;
+    'fileInfos': IFileInfo[];
+    'invalidPathError': string;
+    'ntotalFile': number;
+    'viewingOffset': number;
+  }
+}
+
+export class FileView extends React.PureComponent<IFileViewProps> {
+  constructor(props: IFileViewProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const {
+      allowDownload,
+      beginInputHandler,
+      end,
+      endInputHandler,
+      lastFetched,
+      offset,
+      offsetInputHandler,
+      path,
+      queryStringPrefix,
+      queryStringSuffix,
+      textAreaHeight,
+      viewData
+    } = this.props;
+
+    return (
+      <React.Fragment>
+        <h5>
+          {viewData.currentDirectory ? viewData.currentDirectory.absolutePath : viewData.currentPath}: <small>First 5KB
+          from {viewData.viewingOffset} in ASCII</small>
+        </h5>
+        <Form className="mb-3 viewData-file-form" id="viewDataFileForm" inline={true}>
+          <FormGroup className="mb-2 mr-sm-2 w-100">
```

Comment:
Are these class names standard, and how do we decide which ones to use?

Reply:
Yes, these are part of the bootstrap framework